stage1_fly/run/main.go

```diff
@@ -129,7 +125,24 @@ func evaluateMounts(rfs string, app string, p *stage1commontypes.Pod) ([]flyMoun
 	}
 
 	// Merge command-line Mounts with ImageManifest's MountPoints
-	for _, mp := range manifestMPs {
+	var imAppManifestMPs []types.MountPoint
+	if imApp := p.Images[app].App; imApp != nil {
+		imAppManifestMPs := imApp.MountPoints
+		for _, mp := range imAppManifestMPs {
+			tuple, exists := namedVolumeMounts[mp.Name]
+			switch {
+			case exists && tuple.M.Path != mp.Path:
+				return nil, fmt.Errorf("conflicting path information from mount and mountpoint %q", mp.Name)
```

Comment:
Since you asked for my opinion: it has really not changed. I never wanted `fly` to be a stage1 in the first place exactly because of these kinds of issues and confusion ([eg](https://github.com/coreos/rkt/issues/2064), [eg](https://github.com/coreos/rkt/issues/2792), [eg](https://github.com/coreos/rkt/issues/2577), [eg](https://github.com/coreos/rkt/issues/2846), [eg](https://github.com/coreos/rkt/issues/1889#issuecomment-165761507), [eg](https://github.com/coreos/rkt/pull/2251#issuecomment-193843549, [eg](https://github.com/coreos/rkt/issues/2828#issuecomment-229959039))).

The intention from the beginning of rkt was that stage1s implement pods, according to the spec. `fly` breaks the pod abstraction completely - all bets are off, and it's questionable why it should follow any spec behaviour other than for _some_ consistency with the other stage1s - even though it will always be inherently and fundamentally inconsistent. This is why I [intended](https://github.com/coreos/rkt/pull/1072/files#diff-ed6d1173e544a22028cf90c059ee2219R104) and [felt](https://github.com/coreos/rkt/pull/1833#issuecomment-162078322) (and [still feel](https://github.com/coreos/rkt/issues/2085#issuecomment-185220764)) that it should be [completely distinct from run/pods](https://github.com/coreos/rkt/pull/1833#discussion_r48075247) - otherwise this cognitive dissonance is going to continue to cause confusion and problems.

To get us into a better place I would strongly suggest that the proponents of the current approach find an owner for the following issues and prioritise them post haste:
- https://github.com/coreos/rkt/issues/1897
- https://github.com/coreos/rkt/issues/2085 and https://github.com/coreos/rkt/issues/2792#issuecomment-225956607
- https://github.com/coreos/rkt/issues/1889